galsim/utilities.py

```diff
@@ -418,3 +418,133 @@ def _gammafn(x):
                0.00000000000000000141, -0.00000000000000000023, 0.00000000000000000002
              )
 
+def interleaveImages(im_list,N,offsets,add_flux=True,suppress_warnings=False,catch_offset_errors=True):
+    """
+    Interleaves two or more images and outputs a larger image.
```

Comment:
This needs a lot more explanation at the start of this docstring.  "Interleaving" images is not a standard thing to do to astronomical images, so you need explain what it is doing right at the start.  Here is a straw man for you to edit:

Interleaves the pixel values from two or more images into a single larger image.  This routine is used to convert a list of images taken at a series of dither offsets into a single higher resolution image, where the value in each final pixel is the observed pixel value from exactly one of the original images.  It can be used to build a Nyquist-sampled image from images that were observed with pixels larger than the Nyquist scale.

In the original observed images, the integration of the surface brightness over the pixels is equivalent to convolution by the pixel profile and then sampling at the centers of the pixels.  This procedure thus simulates an observation sampled at a higher resolution than the original images, but still with the same pixel convolution.

One of the advantages of this procedure over other procedures that build a higher resolution image from a set of low resolution images, such as drizzle, is that the noise in the final image is not correlated.  Each pixel value comes from just a single input image.  The noise properties of the pixels are thus preserved.

A significant disadvantage is that the input images must be uniformly spaced in both x and y, which is difficult to achieve with real observations, but which can be precisely achieved with a series of simulated images.